### ISSUE TITLE
Keras ProgbarLogger: OverflowError: cannot convert float infinity to integer

### ISSUE THREAD

**Issue by dfalbel** [tensorflow/tensorflow#38618]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Colab

**Describe the current behavior**

It seems that if, at some time accuracy is equal 0 this code will error and stop training.

With TF 2.2-rc3 the following piece of code returns an error:

```
import tensorflow as tf
data = tf.random.uniform(shape=(1000, 784),  maxval=15)
labels = tf.random.uniform(shape=(1000,10), maxval = 1, dtype=tf.int32)
model = tf.keras.Sequential([
  tf.keras.layers.Dense(32, activation = "relu"),
  tf.keras.layers.Dense(10, activation = "softmax")
])
model.compile(loss = "binary_crossentropy", optimizer="sgd", metrics=["accuracy"])
model.fit(data, labels, callbacks = [tf.keras.callbacks.ProgbarLogger()])
```

> OverflowError: cannot convert float infinity to integer

**Describe the expected behavior**

With current 2.1, no error is returned:

```
import tensorflow as tf
data = tf.random.uniform(shape=(1000, 784),  maxval=15)
labels = tf.random.uniform(shape=(1000,10), maxval = 1, dtype=tf.int32)
model = tf.keras.Sequential([
  tf.keras.layers.Dense(32, activation = "relu"),
  tf.keras.layers.Dense(10, activation = "softmax")
])
model.compile(loss = "binary_crossentropy", optimizer="sgd", metrics=["accuracy"])
model.fit(data, labels, callbacks = [tf.keras.callbacks.ProgbarLogger()])
```

**Standalone code to reproduce the issue**

See colab here for 2.2-rc3 version: https://colab.research.google.com/drive/1Cyu0pBYCYtoxzUs2JPc-oSRr_WG9PNQt

And here with 2.1 working version:

https://colab.research.google.com/drive/1Aq7vhnt91C8MbNP35RUrhszR9xomn-hM

**Comment by gadagashwini** [comment]
Was able to reproduce the issue with Tf2.2.rc3.
Please find the gist [here](https://colab.sandbox.google.com/gist/gadagashwini/616b04d15c11f18a139348b142c17414/untitled513.ipynb). Thanks!

**Comment by gulshanrana10** [comment]
@dfalbel hello, 
If you want to use Keras ProgbarLogger in TensorFlow version 1, Then I would suggest two solution:-
1. Either use **_tf.compat.v1.keras.callbacks.ProgbarLogger( )_**  instead of tf.keras.callbacks.ProgbarLogger( )
or,
2. Provide a parameter   **count_mode='steps'** i.e ProgbarLogger( count_mode='steps') because you will have to look for batches seen

**Comment by dfalbel** [comment]
@gulshanrana10 I don't understand your answer.

1.  I am not using TF 1, I am using 2.2-rc3. 
2. How can I use the `count_mode="samples"`? I don't want to use the `steps` mode. Why does it work in 2.1 and not in 2.2-rc3?

**Comment by dfalbel** [comment]
@gulshanrana10 I don't understand your answer.

1.  I am not using TF 1, I am using 2.2-rc3. 
2. How can I use the `count_mode="samples"`? I don't want to use the `steps` mode. Why does it work in 2.1 and not in 2.2-rc3?

**Comment by dfalbel** [comment]
@gulshanrana10 I don't understand your answer.

1.  I am not using TF 1, I am using 2.2-rc3. 
2. How can I use the `count_mode="samples"`? I don't want to use the `steps` mode. Why does it work in 2.1 and not in 2.2-rc3?

**Comment by omalleyt12** [comment]
@dfalbel Thanks for the issue!

You shouldn't be passing the `ProgbarLogger` directly in `Model.fit`, it is created automatically or not based on the `verbose` argument. This works for me:

```python
import tensorflow as tf
data = tf.random.uniform(shape=(1000, 784),  maxval=15)
labels = tf.random.uniform(shape=(1000,10), maxval = 1, dtype=tf.int32)
model = tf.keras.Sequential([
  tf.keras.layers.Dense(32, activation = "relu"),
  tf.keras.layers.Dense(10, activation = "softmax")
])
model.compile(loss = "binary_crossentropy", optimizer="sgd", metrics=["accuracy"])
model.fit(data, labels, verbose=1)
```

**Comment by dfalbel** [comment]
Fair enough. Although, if it's unusable IMHO it should not be exported.
